scheduler/src/cook/mesos/task.clj

```diff
@@ -280,12 +280,24 @@
   {"DOCKER" :container-type-docker
    "MESOS" :container-type-mesos})
 
+(defn- assign-port-mappings
+  "Assign port mappings from offer. Port n in the cook job should be mapped to the nth
+   port taken from the offer."
+  [container ports-assigned]
+  (update-in container [:docker :port-mappings]
+             (fn [port-mappings]
+               (map (fn [{:keys [host-port] :as port-mapping}]
+                      (if (contains? ports-assigned host-port)
+                        (assoc port-mapping :host-port (ports-assigned host-port))
```

Comment:
Since `ports-assigned` is a vector, it'd be clearer to me if you had used `(get ports-assigned host-port)` or  `(nth ports-assigned host-port)` instead.

Reply:
Done